packages/services/api/src/modules/organization/providers/organization-access-tokens.ts

```diff
@@ -315,31 +569,533 @@ export class OrganizationAccessTokens {
+    const isOTELTracingEnabled = organization.featureFlags.otelTracing;
+
+    return (id: Permission) =>
+      (!isAppDeplymentsEnabled && id.startsWith('appDeployment:')) ||
+      (!isOTELTracingEnabled && id.startsWith('traces:'))
+        ? false
+        : true;
+  }
+
+  async getAvailablePermissionGroupsForOrganization(
+    organization: Organization,
+  ): Promise<Array<PermissionGroup>> {
+    const filter = this.createFeatureFlagPermissionFilter(organization);
+
+    return OrganizationAccessTokensPermissions.permissionGroups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions
+          .filter(permission => filter(permission.id))
+          .map(permission => ({
+            ...permission,
+            isAssignableByViewer: true,
+          })),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  async getAvailablePermissionGroupsForMembership(membership: OrganizationMembership) {
+    const organization = await this.storage.getOrganization({
+      organizationId: membership.organizationId,
+    });
+    const groups = await this.getAvailablePermissionGroupsForOrganization(organization);
+    const memberPermissions = membership.assignedRole.role.allPermissions;
+
+    return groups
+      .map(group => ({
+        ...group,
+        permissions: group.permissions.map(permission => ({
+          ...permission,
+          isAssignableByViewer: memberPermissions.has(permission.id),
+        })),
+      }))
+      .filter(group => group.permissions.length !== 0);
+  }
+
+  async getPermissionsForAccessToken(accessToken: OrganizationAccessToken) {
+    if (!accessToken.userId) {
+      return accessToken.permissions ?? [];
+    }
+    const organization = await this.storage.getOrganization({
+      organizationId: accessToken.organizationId,
+    });
+    const membership = await this.members.findOrganizationMembership({
+      organization,
+      userId: accessToken.userId,
+    });
+    if (!membership) {
+      return [];
+    }
+
+    const allRolePermissions = membership.assignedRole.role.allPermissions;
+
+    if (!accessToken.permissions) {
+      return Array.from(membership.assignedRole.role.allPermissions);
+    }
+
+    return accessToken.permissions.filter(permission => allRolePermissions.has(permission));
+  }
+
+  async getResourceAssignmentsForAccessToken(accessToken: OrganizationAccessToken) {
+    if (accessToken.userId === null) {
+      return this.resourceAssignments.resolveGraphQLMemberResourceAssignment({
+        organizationId: accessToken.organizationId,
+        resources: accessToken.assignedResources,
+      });
+    }
+
+    const organization = await this.storage.getOrganization({
+      organizationId: accessToken.organizationId,
+    });
+    const membership = await this.members.findOrganizationMembership({
+      organization,
+      userId: accessToken.userId,
+    });
+
+    // TODO: we could handle this different TBH
```

Comment:
should we?
Prefer creating a ticket and deleting the line if it matters.

Reply:
removed